### ISSUE TITLE
(Android) Audio distorted when started with dualshock 4 plugged in

### ISSUE THREAD

**Issue by Steveice10** [hrydgard/ppsspp#11917]
When I start PPSSPP with my dualshock 4 plugged in using a USB OTG cable, game audio comes out higher pitched and stuttering.

This only happens when the app is *started* with the controller plugged in; if I unplug the controller without reopening the app, the problem persists, and if I plug the controller in only after starting the app, the problem doesn't occur. My guess is it probably has something to do with the initial buffer/sample rate configuration and how the dualshock 4 provides an audio output over USB. Do note that the audio is still messed up whether I use the internal headphone jack of my phone or the one on the controller, though.

I'm using a Samsung Galaxy S7 edge on Android Oreo, with PPSSPP version 1.8.0.

**Comment by Steveice10** [comment]
Took a look at the log output.

Controller Unplugged:
```
NativeApp.audioInit() -- Using OpenSL audio! frames/buffer: 192	 optimal sr: 48000	 actual sr: 48000
```

Controller Plugged In:
```
NativeApp.audioInit() -- Using OpenSL audio! frames/buffer: 160	 optimal sr: 32000	 actual sr: 32000
...
Invalid sample rate 32000 - choosing 44100
```

**Comment by unknownbrackets** [comment]
Hm, I think we should not necessarily correct it there, or if we do force to 44 we need to do it earlier so SYSPROP_AUDIO_SAMPLE_RATE gives the corrected rate.

The problem must be that SYSPROP_AUDIO_SAMPLE_RATE is still giving 32k.

-[Unknown]

**Comment by Drasikk** [comment]
In which games does that problem?

**Comment by hrydgard** [comment]
That should affect all games. But yes, we should not reject 32000, even if it's not listed as a valid sample rate by the platform (I recall seeing a list somewhere). But I'm not sure if we ever get a notification if the optimal sample rate changes later, so we'll probably simply be stuck at 32000.

**Comment by hrydgard** [comment]
Well, I remembered wrong. We should simply support a lot more sampling rates. Search for SL_SAMPLINGRATE_32 for example in https://www.khronos.org/registry/OpenSL-ES/api/1.0/OpenSLES.h. In fact we should be able to simply multiply the optimal sample rate by 1000 and plug it in, instead of choosing between the defines.

**Comment by unknownbrackets** [comment]
Makes sense, that's easy enough.

-[Unknown]

**Comment by mrchezco1995** [comment]
Just experienced this issue today. Since the DS4 basically also has an audio chip inside (hence the audio Jack on the controller), Android is just confused (or what?) where to route audio and even when headphone connected to the phone itself is distorted. Also add that odd bitrate some here mentioned.

It's simply an easy fix for Android.  Just disable USB Audio Routing on Developer options. Solved :')

**Comment by unknownbrackets** [comment]
Are you trying the 1.8.0 version, or the latest git build?

This issue is closed, but nothing has changed in 1.8.0.  The fix will be in a future release, but you can already see the fix in the latest git build.

-[Unknown]
